tensorflow/tsl/lib/random/random_distributions.h

```diff
@@ -85,56 +128,85 @@ class UniformDistribution<Generator, Eigen::half> {
   }
 };
 
-template <class Generator>
-class UniformDistribution<Generator, bfloat16> {
+template <class Generator, bool Vectorized>
+class UniformDistribution<Generator, bfloat16, Vectorized> {
  public:
   // The number of elements that will be returned.
-  static constexpr int kResultElementCount = Generator::kResultElementCount;
+  // Set the number to be Eigen packet size of type at least, so computations
+  // can be vectorized using SIMD on CPU.
+  static constexpr int kVectorLength = std::max(
+      static_cast<const int>(Eigen::internal::packet_traits<bfloat16>::size),
+      Generator::kResultElementCount);
+  static constexpr int kResultElementCount =
+      Vectorized ? kVectorLength : Generator::kResultElementCount;
   // Cost of generation of a single element (in cycles).
   static constexpr int kElementCost = 3;
   // Indicate that this distribution may take variable number of samples
   // during the runtime.
   static constexpr bool kVariableSamplesPerOutput = false;
   typedef Array<bfloat16, kResultElementCount> ResultType;
   typedef bfloat16 ResultElementType;
+  // Helper definition for the format function.
+  typedef bfloat16 (*FormatFunc)(uint16);
 
   PHILOX_DEVICE_INLINE
   ResultType operator()(Generator* gen) {
+#ifdef __CUDA_ARCH__
+    static_assert(!Vectorized, "Can't vectorize Distribution on GPU");
```

Comment:
Why are we static asserting?  It looks like we're just looping in either case.

Reply:
It's because `Vectorized` flag is passed from op registry and useless on GPU. In case framework developer wrongly set this flag for GPU op, I made this assert here. Do you think it's safe to ignore this flag on GPU and remove this assert?

Reply:
You should remove the `Vectorized` template parameter altogether.  You never partially-specialize based on it.   Always vectorize if possible, otherwise fall back to the non-vectorized version.